src/__lib__/serverSideOnlyComponent.js

```diff
@@ -0,0 +1,29 @@
+// @flow
+import { Component } from 'preact';
+
+const contentReferences = [];
+
+const serverSideOnlyComponent = (MyComponent, contentReference) =>
+    class extends Component {
+        constructor(props) {
+            super(props);
+
+            if (contentReference) {
```

Comment:
what do you think about auto-generating an id, rather than passing one in? is there a value in being able to specify it?

Reply:
Currently this identifier is used for associating the `dom` element with `data.content` so `contentReference` needs to match the key of the Component's source data in `data.content`.